### ISSUE TITLE
[JDA] Message content intent

### ISSUE THREAD

**Issue by kyngs** [aikar/commands#366]
As of first September 2022, Discord **will** require bots to have a special (privileged) intent to read messages, if they are on more than 100 guilds.

Discord's devs have clearly stated, that this intent will not be granted to you if you want to use it for commands (!command something).

This makes the current JDA implementation pretty much unusable.

**Solutions**

There are two solutions I can think of:

1. Leave it like it is and essentially make it available only for private bots. 
2. Redo the JDA implementation with complete slash command support.

In my opinion, the first solution would be unfortunate. However, I am not sure whether the second solution is even possible. 

If you agree, I would do some research and if possible, choose the second solution.
This will break all backward compatibility for applications written for the current JDA implementation.

**Comment by kyngs** [comment]
I've found a few (unfortunate) limitations, I will attempt to keep this list up to date. There might be a workaround for some of these, but I doubt it.

- [ ] After you register all of your commands, you must call propagate() method on the manager. This is because we must (not really, but should) update commands at once.
- [ ] You can only chain subcommands twice. For example: /command sub1 sub2 args is possible, but /command sub1 sub2 sub3 (...) args is not. This will throw an exception when registering
- [ ] I can't really explain it, so I will show it as an example. This is not allowed: Having /command test nom and /command test at the same time.

**Comment by NachtRaben** [comment]
Currently have slash command support with my project over at https://github.com/NachtRaben/ArmaCore with a propagation method. Though realistically I'm not very pleased with the implementation as I was learning ACF at the time of writing it, in theory it's not too hard to add it.

Another thing to be mindful of is the new Slash Permissions V2 discord is rolling out at https://discord.com/blog/slash-commands-permissions-discord-apps-bots which might require finer control with annotations and such to implement more seamlessly with user/role/etc based permissions being added to discord.

**Comment by kyngs** [comment]
> Currently have slash command support with my project over at https://github.com/NachtRaben/ArmaCore with a propagation method. Though realistically I'm not very pleased with the implementation as I was learning ACF at the time of writing it, in theory it's not too hard to add it.
> 
> Another thing to be mindful of is the new Slash Permissions V2 discord is rolling out at https://discord.com/blog/slash-commands-permissions-discord-apps-bots which might require finer control with annotations and such to implement more seamlessly with user/role/etc based permissions being added to discord.
> 
> 

Thanks, I will definitely take a look on it. My implementation is also pretty dirty.

**Comment by kyngs** [comment]
> Currently have slash command support with my project over at https://github.com/NachtRaben/ArmaCore with a propagation method. Though realistically I'm not very pleased with the implementation as I was learning ACF at the time of writing it, in theory it's not too hard to add it.
> 
> Another thing to be mindful of is the new Slash Permissions V2 discord is rolling out at https://discord.com/blog/slash-commands-permissions-discord-apps-bots which might require finer control with annotations and such to implement more seamlessly with user/role/etc based permissions being added to discord.

So I finally found time for this. According to discord, bot developers cannot really control it yet:
![image](https://user-images.githubusercontent.com/38181667/172456825-0eeadae1-4c91-46bc-9cca-96d0b44bf09d.png)

So, perhaps, I should implement the current system and wait before the new one gets made public?

**Comment by NachtRaben** [comment]
Might be some useful information here that users have already gathered.
https://github.com/DV8FromTheWorld/JDA/pull/2113

**Comment by chickeneer** [comment]
I think, the way forward will be to have two modules. A jda module that runs through reading chat (legacy old way).
And another one that registers commands through the slash-command method.

The overall usage would be that the same project could have both versions going through separate command managers.
I think this is a reasonable way to approach this, if you think of a reason it should be handled differently - holler out. :)

**Comment by kyngs** [comment]
> I think, the way forward will be to have two modules. A jda module that runs through reading chat (legacy old way). And another one that registers commands through the slash-command method.
> 
> The overall usage would be that the same project could have both versions going through separate command managers. I think this is a reasonable way to approach this, if you think of a reason it should be handled differently - holler out. :)

Makes sense, in my fork I still kept the old module, so it should be safe to use.
However, it will require migration to the new JDA version.
